app/src/main/java/com/github/steroidteam/todolist/ListSelectionActivity.java

```diff
@@ -1,27 +1,131 @@
 package com.github.steroidteam.todolist;
 
-import androidx.appcompat.app.ActionBar;
-import androidx.appcompat.app.AppCompatActivity;
-import androidx.appcompat.widget.Toolbar;
-
 import android.content.Intent;
 import android.os.Bundle;
+import android.view.LayoutInflater;
 import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.github.steroidteam.todolist.todo.TodoList;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class ListSelectionActivity extends AppCompatActivity {
+    private static todoListAdapter adapter;
+    private ArrayList<TodoList> todoLists;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_list_selection);
+        todoLists = new ArrayList<>();
+        TodoList todoList = new TodoList("Sweng project");
+        todoLists.add(todoList);
+        todoLists.add(new TodoList("Homework"));
+        todoLists.add(new TodoList("Some stuff"));
+        setTitle("TODO Lists");
+
+        ListView listView = findViewById(R.id.activity_list_selection_itemlist);
+        adapter = new todoListAdapter(todoLists);
+        setListViewSettings(listView);
         //Toolbar toolbar = findViewById(R.id.list_selection_toolbar);
         //setSupportActionBar(toolbar);
         //ActionBar actionBar = getSupportActionBar();
         //actionBar.setDisplayHomeAsUpEnabled(true);
     }
 
+    private void setListViewSettings(ListView listView) {
+        listView.setAdapter(adapter);
+        listView.setLongClickable(true);
+
+        listView.setOnItemLongClickListener((adapterView, view, i, l) -> {
+            return true;
+        });
+    }
+
     public void openList(View view){
         Intent itemViewActivity = new Intent(ListSelectionActivity.this, ItemViewActivity.class);
         startActivity(itemViewActivity);
     }
+
+    private class todoListAdapter extends BaseAdapter{
+
+        private List<TodoList> todoLists;
+
+        public todoListAdapter(List<TodoList> todoLists){
+            this.todoLists = todoLists;
+        }
+
+        @Override
+        public int getCount() {
+            return todoLists.size();
+        }
+
+        @Override
+        public Object getItem(int i) {
+            return todoLists.get(i);
+        }
+
+        @Override
+        public long getItemId(int i) {
+            return todoLists.get(i).getId().getLeastSignificantBits();

```

Comment:
Why are we using this instead of the ID? :thinking:

Reply:
Because the ID is an int and I've made some research and it seemed a used way to do but we could certainly change that in the future

Reply:
So just to be clear, is this to cast the ID from `int` to `long`? Because if that's the case, you can just do:

```java
(long) todoLists.get(i).getId()
```

Reply:
well yes it would make sense and it's more understandable I'll do it rn

Reply:
Makes sense, thanks for the clarification!